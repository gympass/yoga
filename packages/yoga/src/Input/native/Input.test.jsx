import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import { ThemeProvider, Input } from '../..';

describe('<Input />', () => {
  describe('Snapshots', () => {
    it('should match with default input', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Input />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match with label', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Input label="Input" />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match when input is focused', () => {
      const { toJSON, getByTestId } = render(
        <ThemeProvider>
          <Input label="Input" testID="input" />
        </ThemeProvider>,
      );

      fireEvent(getByTestId('input'), 'focus');

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match with disabled input', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Input disabled />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match with error', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Input error="Error message" />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match with helper text and max length', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Input helper="Helper text" maxLength={20} />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match with helper text, max length and hideMaxLength', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Input helper="Helper text" maxLength={20} hideMaxLength />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match with full width', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Input label="Label" full />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    it('should call onChangeText', () => {
      const onChangeTextMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Input label="Input" testID="input" onChangeText={onChangeTextMock} />
        </ThemeProvider>,
      );

      fireEvent.changeText(getByTestId('input'), 'foo');

      expect(onChangeTextMock).toHaveBeenCalled();
    });

    it('should not call onChangeText when input is disabled', () => {
      const onChangeTextMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Input
            label="Input"
            testID="input"
            onChangeText={onChangeTextMock}
            disabled
          />
        </ThemeProvider>,
      );

      fireEvent(getByTestId('input'), 'focus');

      expect(onChangeTextMock).not.toHaveBeenCalled();
    });

    it('should call onFocus', () => {
      const onFocusMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Input label="Input" testID="input" onFocus={onFocusMock} />
        </ThemeProvider>,
      );

      fireEvent(getByTestId('input'), 'focus');

      expect(onFocusMock).toHaveBeenCalled();
    });

    it('should call onBlur', () => {
      const onBlurMock = jest.fn();

      const { getByTestId } = render(
        <ThemeProvider>
          <Input label="Input" testID="input" onBlur={onBlurMock} />
        </ThemeProvider>,
      );

      fireEvent(getByTestId('input'), 'focus');
      fireEvent(getByTestId('input'), 'blur');

      expect(onBlurMock).toHaveBeenCalled();
    });
  });

  describe('maxLength', () => {
    it('should update maxLength counter when add character', () => {
      const { getByText, rerender } = render(
        <ThemeProvider>
          <Input label="Input" testID="input" maxLength={10} />
        </ThemeProvider>,
      );

      expect(getByText('0/10')).toBeTruthy();

      rerender(
        <ThemeProvider>
          <Input label="Input" value="foo" maxLength={10} />
        </ThemeProvider>,
      );

      expect(getByText('3/10')).toBeTruthy();
    });
  });

  describe('clean button', () => {
    it('should call onClean when press clean button', () => {
      const onCleanMock = jest.fn();

      const { getByRole } = render(
        <ThemeProvider>
          <Input label="Input" value="foo" onClean={onCleanMock} />
        </ThemeProvider>,
      );

      fireEvent.press(getByRole('button'));

      expect(onCleanMock).toHaveBeenCalledWith('');
    });

    it('should test if clean button is present', () => {
      const { rerender, queryByRole } = render(
        <ThemeProvider>
          <Input label="Input" />
        </ThemeProvider>,
      );

      // closeButton
      expect(queryByRole('button')).toBeNull();

      rerender(
        <ThemeProvider>
          <Input label="Input" value="foo" />
        </ThemeProvider>,
      );

      expect(queryByRole('button')).not.toBeNull();
    });
  });
});
