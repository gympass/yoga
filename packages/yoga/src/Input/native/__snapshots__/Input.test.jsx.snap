// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Snapshots should match when input is focused 1`] = `
<View
  full={false}
  style={
    Array [
      Object {
        "position": "relative",
        "width": 320,
      },
      Object {},
    ]
  }
>
  <TextInput
    cleanable={true}
    disabled={false}
    editable={true}
    focus={true}
    full={false}
    onBlur={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#231B22",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 14,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 52,
          "paddingTop": 16,
          "width": 320,
        },
        Object {
          "height": 52,
        },
      ]
    }
    testID="input"
    typed={false}
    value=""
  />
  <View
    collapsable={false}
    focus={true}
    pointerEvents="none"
    style={
      Object {
        "left": 14,
        "position": "absolute",
        "top": 24,
        "transform": Array [
          Object {
            "translateY": -8,
          },
        ],
      }
    }
    typed={false}
  >
    <Text
      collapsable={false}
      disabled={false}
      focus={true}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 14,
          "paddingLeft": 4,
          "paddingRight": 4,
        }
      }
      typed={false}
    >
      Input
    </Text>
  </View>
</View>
`;

exports[`<Input /> Snapshots should match with default input 1`] = `
<View
  full={false}
  style={
    Array [
      Object {
        "position": "relative",
        "width": 320,
      },
      Object {},
    ]
  }
>
  <TextInput
    cleanable={true}
    disabled={false}
    editable={true}
    focus={false}
    full={false}
    onBlur={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D7D7E0",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#6B6B78",
          "fontFamily": "Rubik",
          "fontSize": 14,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 52,
          "paddingTop": 16,
          "width": 320,
        },
        Object {
          "height": 52,
        },
      ]
    }
    typed={false}
    value=""
  />
</View>
`;

exports[`<Input /> Snapshots should match with disabled input 1`] = `
<View
  full={false}
  style={
    Array [
      Object {
        "position": "relative",
        "width": 320,
      },
      Object {},
    ]
  }
>
  <TextInput
    cleanable={true}
    disabled={true}
    editable={false}
    focus={false}
    full={false}
    onBlur={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D7D7E0",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#D7D7E0",
          "fontFamily": "Rubik",
          "fontSize": 14,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 52,
          "paddingTop": 16,
          "width": 320,
        },
        Object {
          "height": 52,
        },
      ]
    }
    typed={false}
    value=""
  />
</View>
`;

exports[`<Input /> Snapshots should match with error 1`] = `
<View
  full={false}
  style={
    Array [
      Object {
        "position": "relative",
        "width": 320,
      },
      Object {},
    ]
  }
>
  <TextInput
    cleanable={true}
    disabled={false}
    editable={true}
    error="Error message"
    focus={false}
    full={false}
    onBlur={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#CA4808",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#6B6B78",
          "fontFamily": "Rubik",
          "fontSize": 14,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 52,
          "paddingTop": 16,
          "width": 320,
        },
        Object {
          "height": 52,
        },
      ]
    }
    typed={false}
    value=""
  />
  <View
    disabled={false}
    full={false}
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 4,
          "maxWidth": 320,
          "width": 320,
        },
      ]
    }
  >
    <Text
      disabled={false}
      error="Error message"
      focused={false}
      style={
        Array [
          Object {
            "color": "#CA4808",
            "flexWrap": "wrap",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "fontWeight": "400",
          },
        ]
      }
    >
      Error message
    </Text>
  </View>
</View>
`;

exports[`<Input /> Snapshots should match with full width 1`] = `
<View
  full={true}
  style={
    Array [
      Object {
        "position": "relative",
        "width": "100%",
      },
      Object {},
    ]
  }
>
  <TextInput
    cleanable={true}
    disabled={false}
    editable={true}
    focus={false}
    full={true}
    onBlur={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D7D7E0",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#6B6B78",
          "fontFamily": "Rubik",
          "fontSize": 14,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 52,
          "paddingTop": 16,
          "width": "100%",
        },
        Object {
          "height": 52,
        },
      ]
    }
    typed={false}
    value=""
  />
  <View
    collapsable={false}
    focus={false}
    pointerEvents="none"
    style={
      Object {
        "left": 16,
        "position": "absolute",
        "top": 24,
        "transform": Array [
          Object {
            "translateY": -8,
          },
        ],
      }
    }
    typed={false}
  >
    <Text
      collapsable={false}
      disabled={false}
      focus={false}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "color": "#6B6B78",
          "fontFamily": "Rubik",
          "fontSize": 14,
        }
      }
      typed={false}
    >
      Label
    </Text>
  </View>
</View>
`;

exports[`<Input /> Snapshots should match with helper text and max length 1`] = `
<View
  full={false}
  style={
    Array [
      Object {
        "position": "relative",
        "width": 320,
      },
      Object {},
    ]
  }
>
  <TextInput
    cleanable={true}
    disabled={false}
    editable={true}
    focus={false}
    full={false}
    maxLength={20}
    onBlur={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D7D7E0",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#6B6B78",
          "fontFamily": "Rubik",
          "fontSize": 14,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 52,
          "paddingTop": 16,
          "width": 320,
        },
        Object {
          "height": 52,
        },
      ]
    }
    typed={false}
    value=""
  />
  <View
    disabled={false}
    full={false}
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 4,
          "maxWidth": 320,
          "width": 320,
        },
      ]
    }
  >
    <Text
      disabled={false}
      focused={false}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "flexWrap": "wrap",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "fontWeight": "400",
          },
        ]
      }
    >
      Helper text
    </Text>
    <Text
      disabled={false}
      focused={false}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "flexWrap": "wrap",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "fontWeight": "400",
          },
        ]
      }
    >
      0
      /
      20
    </Text>
  </View>
</View>
`;

exports[`<Input /> Snapshots should match with helper text, max length and hideMaxLength 1`] = `
<View
  full={false}
  style={
    Array [
      Object {
        "position": "relative",
        "width": 320,
      },
      Object {},
    ]
  }
>
  <TextInput
    cleanable={true}
    disabled={false}
    editable={true}
    focus={false}
    full={false}
    maxLength={20}
    onBlur={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D7D7E0",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#6B6B78",
          "fontFamily": "Rubik",
          "fontSize": 14,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 52,
          "paddingTop": 16,
          "width": 320,
        },
        Object {
          "height": 52,
        },
      ]
    }
    typed={false}
    value=""
  />
  <View
    disabled={false}
    full={false}
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 4,
          "maxWidth": 320,
          "width": 320,
        },
      ]
    }
  >
    <Text
      disabled={false}
      focused={false}
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "flexWrap": "wrap",
            "fontFamily": "Rubik",
            "fontSize": 12,
            "fontWeight": "400",
          },
        ]
      }
    >
      Helper text
    </Text>
  </View>
</View>
`;

exports[`<Input /> Snapshots should match with label 1`] = `
<View
  full={false}
  style={
    Array [
      Object {
        "position": "relative",
        "width": 320,
      },
      Object {},
    ]
  }
>
  <TextInput
    cleanable={true}
    disabled={false}
    editable={true}
    focus={false}
    full={false}
    onBlur={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#D7D7E0",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#6B6B78",
          "fontFamily": "Rubik",
          "fontSize": 14,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 52,
          "paddingTop": 16,
          "width": 320,
        },
        Object {
          "height": 52,
        },
      ]
    }
    typed={false}
    value=""
  />
  <View
    collapsable={false}
    focus={false}
    pointerEvents="none"
    style={
      Object {
        "left": 16,
        "position": "absolute",
        "top": 24,
        "transform": Array [
          Object {
            "translateY": -8,
          },
        ],
      }
    }
    typed={false}
  >
    <Text
      collapsable={false}
      disabled={false}
      focus={false}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "color": "#6B6B78",
          "fontFamily": "Rubik",
          "fontSize": 14,
        }
      }
      typed={false}
    >
      Input
    </Text>
  </View>
</View>
`;
